tests/test_effect_queue.py

```diff
@@ -591,65 +591,359 @@ def test_scorpion(self):
         back one faints
         """
-        pass
+        # Kills back 3 pets
+        ref_team = Team([], battle=True)
+        t0 = Team(["sloth", "sloth", "sloth"])
+        t1 = Team(["crocodile"])
+        t1[0].pet.level = 3
+        b = run_sob(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_sob(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
+        # hits back pet 3 times
+        ref_team = Team(["sloth"], battle=True)
+        t0 = Team(["sloth", "sloth"])
+        t0[1].pet._health = (
+            3 * data["pets"]["pet-crocodile"]["level3Ability"]["effect"]["amount"]
+        )
+        t1 = Team(["crocodile"])
+        t1[0].pet.level = 3
+        b = run_sob(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_sob(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
 
     def test_rat(self):
-        pass
+        """Test rat spawns dirty rat on opposing team"""
+        # Summons dirty rat
+        ref_team = Team(["dirty-rat"], battle=True)
+        t0 = Team(["sloth"])
+        t1 = Team(["rat"])
+        t1[0].obj._attack = 1
+        t1[0].obj._health = 1
+        b = run_battle(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_battle(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
+        # Summons multiple dirty rats
+        ref_team = Team(["dirty-rat", "dirty-rat", "dirty-rat"], battle=True)
+        t0 = Team(["sloth"])
+        t1 = Team(["rat"])
+        t1[0].obj._attack = 1
+        t1[0].obj._health = 1
+        t1[0].obj.level = 3
+        b = run_battle(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_battle(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
+        # Summons dirty rats at front of opponent team
+        ref_team = Team(["dirty-rat", "dirty-rat", "sloth"], battle=True)
+        t0 = Team(["sloth", "sloth"])
+        t1 = Team(["rat"])
+        t1[0].obj._attack = 1
+        t1[0].obj._health = 1
+        t1[0].obj.level = 2
+        b = run_battle(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_battle(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
 
     def test_gorilla(self):
-        pass
+        ref_team = Team(["gorilla"], battle=True)
+        ref_team[0].obj._attack = 50
+        ref_team[0].obj._health = 1
+        ref_team[0].obj.ability_counter = 1
+        t0 = Team(["gorilla"])
+        t0[0].obj._attack = 50
+        t0[0].obj._health = 2
+        t1 = Team(["sloth", "sloth"])
+        t1[1].obj._attack = 50
+        b = run_battle(t0, t1)
+        self.assertEqual(b.t0.state, ref_team.state)
+        b = run_battle(t1, t0)
+        self.assertEqual(b.t1.state, ref_team.state)
 
     def test_tiger(self):
         """
         Test tiger for nearly all pets
         """
-        pass
+        self.skipTest("TODO")
```

Comment:
Could you describe the issue with the tiger briefly?

Reply:
Tiger causes a crash when it is behind any other pet with an ability. Have added an example crash after the skip test.